### PR TITLE
Relicense to LGPL

### PR DESCRIPTION
**Pull request by Lysxia**
My understanding is there are no significant issues with switching to a less permissive license, but just as a matter of courtesy I'm pinging you contributors @liyishuai @gares @jjhugues to notify you of what's going on.

I just copied the `Uint63` module from Coq, so IIUC the whole library should be relicensed to LGPL. I heard there are plans to relicense Coq under a weaker license (MPL), and when that happens I'll probably revert back to MIT. The main reasons for copying this module are to recompile it without `-rectypes` (that is somehow used by Coq), and to get it from a stable and less awkward-sounding place than `coq-core.kernel`.

### CODE REVIEW

File: dune-project [hunk 1]
```diff
@@ -7,7 +7,7 @@
 (generate_opam_files true)
 
 (source (github Lysxia/coq-simple-io))
-(license MIT)
+(license LGPL)
```
Comment (liyishuai) [inline dune-project:10]: This should be a SPDX identifier.
```suggestion
(license LGPL-2.1-only)
```
I'm still asking for instructions: https://coq.zulipchat.com/#narrow/stream/237663-coq-community-devs-.26-users/topic/Advising.20against.20licenses.3F/near/279029933

**Review by liyishuai — CHANGES_REQUESTED** [review]

**Review by liyishuai — DISMISSED** [review]
Approving license change as a contributor; provided we figure out it's appropriate to do so.

**Comment by liyishuai** [comment]
Based on the Zulip discussion, I'd propose licensing `Uint63`-related modules under LGPL, and declare the remainder of this library as dynamically linked to `Uint63`. Our minimal change is to add a copyright notice under the `extraction` directory.

**Comment by gares** [comment]
My contribution is minor, do as you like

**Comment by jjhugues** [comment]
My contribution is also a trivial one liner. No objection

**Pull request closed without merging**